libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -241,16 +241,19 @@ private static PolicyManager createPolicyManager() {
             )
         );
 
-        Path trustStorePath = trustStorePath();
-        if (trustStorePath != null) {
+        // conditionally add FIPS entitlements if FIPS only functionality is enforced
+        if ("true".equals(System.getProperty("org.bouncycastle.fips.approved_only"))) {
```

Comment:
You can use `Booleans.parseBoolean` (in ES core lib)

Reply:
👍